dashboard/app/serializers/api/v1/pd/workshop_enrollment_serializer.rb

```diff
@@ -11,12 +11,17 @@ def user_id
     user ? user.id : nil
   end
 
-  def alternate_email
+  def application_id
     application_id = object.try(:application_id)
+    return unless application_id && Pd::Application::TeacherApplication.exists?(id: application_id)
+    application_id
```

Comment:
Is there a cleaner way of writing this?

Reply:
if `object` is a `Pd::Enrollment`, then this whole method could be
```
object&.application&.id
```
`application` will return nil if there is no application, or if the application_id does not point to a valid application.

Reply:
Brilliant! This is much better––just updated it (in both places)

Reply:
I must admit, this feels like magic––how does it know the application that's linked to this enrollment? Is it because of application_id, even though we don't explicitly say it's a foreign key for the `Pd::Application::ApplicationBase` class?

Reply:
whoa. I had assumed it was a `:application` association on enrollment, and I tried it out on the console to make sure that enrollment.application.id did what I expected. but, there is no such association! it turns out there is this method:
```
https://github.com/code-dot-org/code-dot-org/blob/ab1c7b6fadaabb656c54289854f9f32b2e846227/dashboard/app/models/pd/enrollment.rb#L345-L348
```
TBH I'm not sure why there isn't just a `belongs_to :application` association instead of this method 🤷

Reply:
Ah, yes! Then I'm a bit surprised the `&.application&.id` works on an application that doesn't exist –– I suppose the `.find_by` throws an error, but the safe chaining catches that ... Thanks for the sleuthing! Agreed: a `belongs_to` association would make sense here

Reply:
`find_by` doesn't raise, but `find_by!` does:  https://guides.rubyonrails.org/active_record_querying.html#find-by